File: DatosPersonas.py
from Estudiante import Estudiante
from Docente import Docente
from PAAE import PAAE

class DatosPersonas:
    def __init__(self):
        self.Estudiantes = None
        self.Docentes = None
        self.PAAEs = None

    def ingresar_datos_persona(self, tipo_persona):
        ID = input("ID: ")
        Nombre = input("Nombre: ")
        CURP = input("CURP: ")
        Domicilio = input("Domicilio: ")

        if tipo_persona == 1:
            self.Estudiantes = Estudiante(ID, Nombre, CURP, Domicilio)
        elif tipo_persona == 2:
            self.Docentes = Docente(ID, Nombre, CURP, Domicilio)
        elif tipo_persona == 3:
            self.PAAEs = PAAE(ID, Nombre, CURP, Domicilio)

    def obtener_informacion_persona(self, tipo_persona):
        persona_seleccionada = None

        if tipo_persona == 1:
            persona_seleccionada = self.Estudiantes
        elif tipo_persona == 2:
            persona_seleccionada = self.Docentes
        elif tipo_persona == 3:
            persona_seleccionada = self.PAAEs
        else:
            persona_seleccionada = None

        if persona_seleccionada is not None:
            print("\nLa información de la persona seleccionada es:")
            print("ID: ", persona_seleccionada.getID())
            print("Nombre: ", persona_seleccionada.getNombre())
            print("CURP: ", persona_seleccionada.getCURP())
            print("Domicilio: ", persona_seleccionada.getDomicilio())
        else:
            print("Esa opción no es válida")

    def ejecutar(self):
        otra_persona = True

        print("B I E N V E N I D O")
        while otra_persona:
            print("Seleccione la persona a la que desea agregar datos:")
            print("1. Estudiante")
            print("2. Docente")
            print("3. PAAE")
            opc = int(input())

            print("Ingrese los datos de la persona:")
            self.ingresar_datos_persona(opc)

            print("\n¿Desea agregar otro tipo de persona?")
            print("Presione 1 para sí y 0 para no")
            resp = input()
            otra_persona = resp == "1"

        print("\nM E N U")
        print("Seleccione la opción que desee")
        print("1. Obtener información estudiante")
        print("2. Obtener información docente")
        print("3. Obtener información PAAE")
        opc_menu = int(input())

        self.obtener_informacion_persona(opc_menu)

datos_personas = DatosPersonas()
datos_personas.ejecutar()

File: DatosPersonas.java
import java.util.Scanner;

public class DatosPersonas {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // Variables para almacenar los datos ingresados por el usuario
        String ID;
        String Nombre;
        String CURP;
        String Domicilio;

        // Variables para almacenar los datos de las personas
        Persona Estudiantes = null;
        Persona Docentes = null;
        Persona PAAEs = null;
        // Variable para controlar si se desea agregar datos de otra persona
        boolean otraPersona = true;

        System.out.println("B I E N V E N I D O");
        while (otraPersona) {
            System.out.println("Seleccione la persona a la que desea agregar datos:");
            System.out.println("1. Estudiante");
            System.out.println("2. Docente");
            System.out.println("3. PAAE");
            int opc = scanner.nextInt();

            // Pedimos los datos de la persona
            System.out.println("Ingrese los datos de la persona:");
            System.out.print("ID: ");
            scanner.nextLine();
            ID = scanner.nextLine();
            System.out.print("Nombre: ");
            Nombre = scanner.nextLine();
            System.out.print("CURP: ");
            CURP = scanner.nextLine();
            System.out.print("Domicilio: ");
            Domicilio = scanner.nextLine();

            // creamoas la instancia de la opcion seleccionada
            switch (opc) {
                case 1:
                    Estudiantes = new Estudiante(ID, Nombre, CURP, Domicilio);
                    break;
                case 2:
                    Docentes = new Docente(ID, Nombre, CURP, Domicilio);
                    break;
                case 3:
                    PAAEs = new PAAE(ID, Nombre, CURP, Domicilio);
                    break;
                default:
                    break;
            }

            // Preguntamos si quiere a agregar a otra persona
            System.out.println("\n ¿Desea agregar otro tipo de persona?");
            System.out.println("Presione 1 para si y 0 para no");
            String resp = scanner.nextLine();
            otraPersona = resp.equals("1");
        }

        System.out.println("\nM E N U");
        System.out.println("Seleccione la opcion que desee");
        System.out.println("1. Obtener informacion estudiante");
        System.out.println("2. Obtener información docente");
        System.out.println("3. Obtener información PAAE");
        int opcMenu = scanner.nextInt();

        // obtener la informacion de la persona seleccionada
        Persona personSelect = null;
        switch (opcMenu) {
            case 1:
                personSelect = Estudiantes;
                break;
            case 2:
                personSelect = Docentes;
                break;
            case 3:
                personSelect = PAAEs;
                break;
            default:
                break;
        }

        // Mostramos la informacion de la persona seleccionada
        if (personSelect != null) {
            System.out.println("\nLa informacion de la persona que seleccionó es: ");
            System.out.println("ID: " + personSelect.getID());
            System.out.println("Nombre: " + personSelect.getNombre());
            System.out.println("CURP: " + personSelect.getCURP());
            System.out.println("Domicilio: " + personSelect.getDomicilio());
        } else {
            System.out.println("Esa opción no es válida");
        }
    }
}
